.github/workflows/apply-terraform.yml

```diff
@@ -94,22 +105,32 @@ jobs:
       env: 
         INFRASTRUCTURE_FOLDER: ${{ inputs.INFRASTRUCTURE_FOLDER }}
         ENVIRONMENT: ${{ inputs.ENVIRONMENT }}
+        IMAGE_TAG: ${{ inputs.IMAGE_TAG }}
       run: |
         cd main
         tfParams=()
         [[ $INFRASTRUCTURE_FOLDER ]] && tfParams+=( "INFRASTRUCTURE_FOLDER=$INFRASTRUCTURE_FOLDER" )
         tfParams+=( "env=$ENVIRONMENT" )
         make tf_set_or_create_workspace "${tfParams[@]}"
-        make tf_auto_apply "${tfParams[@]}" > /dev/null 2>&1 # TODO: mask any sensitive values
+        if [[ $IMAGE_TAG ]]; then
+          make tf_auto_apply IMAGE_TAG=${{ inputs.IMAGE_TAG }} "${tfParams[@]}" > /dev/null 2>&1 # TODO: mask any sensitive values
```

Comment:
Again, not sure that we need this as a variable, the makefile by deafult uses the commit hash already. And given that `IMAGE_TAG` is defined as the commit hash in the makefile, I'm not sure passing in a variable of the same name will work (it might work and overwrite it but not sure)

Reply:
Discussed offline

For posterity:
this is used to override the image tag defined by default to be the `HEAD` of the current branch (by default main). I override it here so that the user can decide exactly which commit to deploy for rolling back etc